ledger/block/src/transactions/rejected/serialize.rs

```diff
@@ -47,6 +62,12 @@ impl<'de, N: Network> Deserialize<'de> for Rejected<N> {
                 // Parse the rejected transaction from a string into a value.
                 let mut object = serde_json::Value::deserialize(deserializer)?;
 
+                // Try to parse the unconfirmed ID.
+                let contains_unconfirmed_id = object.get("unconfirmed_id").is_some();
+                let unconfirmed_id: Option<N::TransactionID> = contains_unconfirmed_id
+                    .then(|| DeserializeExt::take_from_value::<D>(&mut object, "unconfirmed_id"))
+                    .transpose()?;
```

Comment:
Style option: this might be a little clearer? 
```suggestion
                let unconfirmed_id: Option<N::TransactionID> = if object.get("unconfirmed_id").is_some() {
                    Some(DeserializeExt::take_from_value::<D>(&mut object, "unconfirmed_id")?)
                } else {
                    None
                };

```

Reply:
I was so proud of this one Niklas.